src/netbase.h

```diff
@@ -37,6 +38,79 @@ class proxyType
     bool randomize_credentials;
 };
 
+/**
+ * Convert milliseconds to a struct timeval for e.g. select.
+ */
+struct timeval MillisToTimeval(int64_t nTimeout);
+
+// Represents a read/write socket.
+class MockableSocket
+{
+public:
+    virtual ssize_t send(const char* buf, size_t len, int flags) = 0;
+
+    virtual ssize_t recv(char* buf, size_t len, int flags) = 0;
+
+#ifdef USE_POLL
+    virtual int poll(short events, int timeout_ms) = 0;
+#endif
+
+    virtual int select(bool watch_read, bool watch_write, int timeout_ms) = 0;
+
+    virtual bool IsSelectable() const = 0;
+};
+
+// Represents a non-owned SOCKET.
+//
+// The underlying SOCKET will NOT be closed automatically on object destruction.
+class Socket : public MockableSocket
+{
+    SOCKET m_socket = INVALID_SOCKET;
+
+public:
+    explicit Socket(SOCKET socket) : m_socket{socket}
+    {
+    }
+
+    // Send bytes on the socket. See send(2).
+    ssize_t send(const char* buf, size_t len, int flags) override;
+
+    // Receive bytes from the socket. See recv(2).
+    ssize_t recv(char* buf, size_t len, int flags) override;
+
+#ifdef USE_POLL
+    // Simplified poll(2) interface: Wait until the socket becomes ready for
+    // reading, writing or other events. Timeout after timeout_ms milliseconds.
+    // See poll(2) for return values.
+    //
+    // The input parameter field is a bit mask specifying the event(s) the
+    // application is interested in.
+    //
+    // The bits that may be set in events are defined in <poll.h>. Examples
+    // include:
+    // * POLLIN: There is data to read.
+    // * POLLOUT: Writing is now possible, though a write larger that the
+    //   available space in a socket will still block (unless O_NONBLOCK is
+    //   set).
+    int poll(short events, int timeout_ms) override;
+#endif
+
+    // Simplified select(2) interface: Wait until the socket becomes ready for
+    // reading (if watch_read) and/or writing (if watch_write). Timeout after
+    // timeout_ms milliseconds. See select(2) for return values.
+    int select(bool watch_read, bool watch_write, int timeout_ms) override;
```

Comment:
This could be in the `#else` of `USE_POLL`?

Reply:
Good idea! Now addressed. Thanks!